### PR TITLE
Add admin menus cli

### PR DESCRIPTION
**Pull request by dingo-d**
The first commit kinda violates the SRP of this PR, but it's a minor addition that helps save the trees.

I've added the templates for generating admin menus and submenus, with an example for usage purposes.

The things that are missing are

- [x] Tests
- [x] CLI for generating menus and submenus using these templates

I also noticed 2 minor phpstan issues that should be fixed but didn't want to add them here to prevent further pollution of the PR.

### CODE REVIEW

File: src/AdminMenus/AbstractAdminMenu.php [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+	 * 10  - below Media
+	 * 15  - below Links
+	 * 20  - below Pages
+	 * 25  - below comments
+	 * 60  - below first separator
+	 * 65  - below Plugins
+	 * 70  - below Users
+	 * 75  - below Tools
+	 * 80  - below Settings
+	 * 100 - below second separator
```
Comment (IvanGrginovInf) [inline src/AdminMenus/AbstractAdminMenu.php:204]: nice!

**Review by IvanGrginovInf — CHANGES_REQUESTED** [review]
Awesome! Please fix the existing failed tests and add CLI / Example tests for these classes as well.

File: src/AdminMenus/AbstractAdminMenu.php [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+/**
+ * Abstract class AbstractAdminMenu class.
+ *
+ * Class responsible for creating admin menus, separately from CPT admin menus.
+ */
+abstract class AbstractAdminMenu implements ServiceInterface, RenderableBlockInterface
+{
+
+	/**
+	 * Register all the hooks
```
Comment (volfkarlo) [inline src/AdminMenus/AbstractAdminMenu.php:26]:
```suggestion
	 * Register all the hooks.
```

File: src/AdminMenus/AbstractAdminSubMenu.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+/**
+ * Abstract class AbstractAdminSubMenu class.
+ *
+ * Class responsible for creating admin sub menus.
+ */
+abstract class AbstractAdminSubMenu extends AbstractAdminMenu
+{
+
+	/**
+	 * Register all the hooks
```
Comment (volfkarlo) [inline src/AdminMenus/AbstractAdminSubMenu.php:22]:
```suggestion
	 * Register all the hooks.
```

File: src/AdminMenus/AdminMenuExample.php [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+use EightshiftLibs\AdminMenus\AbstractAdminMenu;
+
+/**
+ * BlockPatternExample class.
+ */
+class AdminMenuExample extends AbstractAdminMenu
+{
+	/**
+	 * Capability for this post type
```
Comment (volfkarlo) [inline src/AdminMenus/AdminMenuExample.php:21]:
```suggestion
	 * Capability for this post type.
```

**Review by volfkarlo — CHANGES_REQUESTED** [review]
Just a few dots at the end of the sentence! 👏

File: src/AdminMenus/AbstractAdminMenu.php (hunk 2 above)
Comment (dingo-d) [reply src/AdminMenus/AbstractAdminMenu.php]: I'll add your suggestions, but the summary in a docblock doesn't have to end with a full stop.

https://docs.phpdoc.org/3.0/guide/references/phpdoc/basic-syntax.html#summary

**Comment by dingo-d** [comment]
@IvanGrginovInf once I merge the fix PR I'll rebase and the issues should be gone 😄

File: src/AdminMenus/AbstractAdminMenu.php (hunk 2 above)
Comment (IvanGrginovInf) [reply src/AdminMenus/AbstractAdminMenu.php]: I'd say 99% of our docblocks don't end in a dot ATM, don't see a need for them to either.

**Review by goranalkovic-infinum — APPROVED** [review]

**Review by IvanGrginovInf — APPROVED** [review]

**Pull request merged**